docker/cmsmon-hadoop-base/Dockerfile

```diff
@@ -21,43 +21,57 @@ ENV PATH $PATH:/usr/hdp/hadoop/bin:/usr/hdp/sqoop/bin
-    libaio libaio-devel net-tools lsof bind-utils initscripts patch \
-    voms-devel globus-gsi-credential-devel globus-gsi-cert-utils-devel \
-    globus-common-devel globus-gsi-sysconfig-devel globus-gsi-callback-devel \
-    oracle-instantclient-tnsnames.ora HEP_OSlibs python-pip  \
-    hadoop-bin-${HADOOP_VERSION} \
-    spark-bin-${SPARK_VERSION} \
-    sqoop-bin-1.4 \
-    hbase-bin-2.3 \
-    cern-hadoop-xrootd-connector cern-hadoop-config &&  \
+RUN yum -y update && \
+    yum install -y  \
+        cern-get-certificate fetch-crl \
+        git-core zip unzip which file bzip2 e2fsprogs e2fsprogs-libs compat-libstdc++-33 \
+        CERN-CA-certs ca-certificates dummy-ca-certs ca-policy-lcg ca-policy-egi-core \
+        ca_EG-GRID ca_CERN-GridCA ca_CERN-LCG-IOTA-CA ca_CERN-Root-2 \
+        wlcg-voms-cms krb5-workstation krb5-libs pam_krb5 myproxy voms-clients-cpp voms-clients-java \
+        sudo openssl openssl-devel openssl-libs openssh openssh-clients python-backports-ssl_match_hostname \
+        cmake voms voms-devel globus-gsi-credential-devel globus-gsi-cert-utils-devel \
+        globus-common-devel globus-gsi-sysconfig globus-gsi-sysconfig-devel globus-gsi-callback-devel \
+        globus-gsi-openssl-error globus-gsi-proxy-ssl globus-openssl-module \
+        shibboleth log4shib xmltooling-schemas opensaml-schemas \
+        perl-Thread-Queue zsh tk freetype perl-ExtUtils-Embed fontconfig \
+        perl-Test-Harness perl-Data-Dumper perl-Digest-MD5 perl-Switch perl-Env \
+        libX11-devel libX11 libXmu libSM libICE libXcursor libXext libXrandr libXft \
+        mesa-libGLU mesa-libGL libXi libXinerama libXft-devel libXrender libXpm \
+        libXpm-devel libXext-devel mesa-libGLU-devel \
+        libaio libaio-devel net-tools lsof bind-utils initscripts patch \
+        voms-devel globus-gsi-credential-devel globus-gsi-cert-utils-devel \
+        globus-common-devel globus-gsi-sysconfig-devel globus-gsi-callback-devel \
+        oracle-instantclient-tnsnames.ora HEP_OSlibs python-pip libffi-devel \
+        hadoop-bin-${HADOOP_VERSION} \
+        spark-bin-${SPARK_VERSION} \
+        sqoop-bin-1.4 \
+        cern-hadoop-xrootd-connector  \
+        cern-hadoop-config && \
+    yum clean all && rm -rf /var/cache/yum && \
     hadoop-set-default-conf.sh analytix && \
     source hadoop-setconf.sh analytix ${HADOOP_VERSION} ${SPARK_MAJOR_VERSION} && \
-    yum clean all && rm -rf /var/cache/yum &&  \
     touch /etc/hadoop/conf/topology.table.file && ln -s /bin/bash /usr/bin/bashs && \
     echo "32 */6 * * * root ! /usr/sbin/fetch-crl -q -r 360" > /etc/cron.d/fetch-crl-docker
-# add fetch-crl to fetch all sertificates
+    # add fetch-crl to fetch all sertificates
 
-# Copy entrypoint.sh according to spark version
 RUN if [ "$SPARK_MAJOR_VERSION" = "spark3" ] ; then \
-        cp /usr/hdp/spark3/kubernetes/dockerfiles/spark/entrypoint.sh /usr/bin/entrypoint.sh ; \
+# build Python3.9 for Centos7 if spark3
```

Comment:
@mrceyhun I suggest to move this python build into separate Dockerfile such that you may re-use it across images. This will also allow to upgrade other Dockerfiles easily, since someone can rebuild python image and re-use it here. I suggest that you'll create new area, e.g. cmsmon-python, and maintain it separately. Then, over here you can use this image and copy python installation (but for that you should make python installation into custom area).

Reply:
Thanks for the suggestion @vkuznet . I created a new docker image `cmsmon-py` and used it as base image in `cmsmon-hadoop-base`. I searched in the community if it is possible to make multistage build and copy python3 files from the builder. It seems it's not possible and not suggested by [the community](https://stackoverflow.com/questions/71053803/how-to-copy-python-from-one-stage-to-another-during-a-multi-stage-docker-build). That's why i used it as base image.

`cmsmon-py` builts latest LCG release python3 version to be compatible with cern, and current latest LCG py3 version is python3.9 which is important for spark jobs.